src/pyinstaller_versionfile/__main__.py

```diff
@@ -1,24 +1,46 @@
 """
 Main file for pyinstaller-versionfile, which is the entrypoint for the command line script.
 """
-
+from pathlib import Path
 import argparse
+import sys
 
+from pyinstaller_versionfile import exceptions
 import pyinstaller_versionfile
 
 
 def main(args=None):
     args = args or parse_args(args)
-    pyinstaller_versionfile.create_versionfile_from_input_file(
-        output_file=args.outfile,
-        input_file=args.metadata_file,
-        version=args.version
-    )
-
+    data = args.metadata_source
+
+    if args.source_format == "yaml":
+        # from_yaml
+        try:
+            src_file = Path(data)
+        except NotImplementedError as err:
+            raise exceptions.InputError(f"Path {src_file.as_posix()} is invalid") from err
+        else:
+            pyinstaller_versionfile.create_versionfile_from_input_file(
+                output_file=args.outfile,
+                input_file=src_file.as_posix(),
+                version=args.version
+            )
+            sys.exit(0)
+    elif args.source_format in ['distribution', 'dist']:
+        # from_distribution
+        pyinstaller_versionfile.create_versionfile_from_distribution(
+            output_file=args.outfile,
+            distname=data,
+            version=args.version
+        )
+        sys.exit(0)
+    else:
+        raise exceptions.InputError(f"Unknown source type {args.source_format}")
```

Comment:
It might be useful listing the supported values for `source_format` here.
Same goes for the help message for the CLI, so having a list with valid options defined as constant would be good.

Reply:
Because of the encapsulated parsing in the current implementation, I think the best way is to set “choices” in the parser argument and to use a built-in check.